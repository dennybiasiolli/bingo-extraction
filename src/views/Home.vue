<template>
  <v-container fluid>
    <DisplayNumbers title="Available numbers" :numbers="availableNumbers" />
  </v-container>
</template>

<script>
import DisplayNumbers from '@/components/DisplayNumbers.vue';

export default {
  name: 'Home',
  components: {
    DisplayNumbers,
  },
  data() {
    return {
      availableNumbers: [...Array(90).keys()].map((i) => i + 1),
    };
  },
};
</script>
