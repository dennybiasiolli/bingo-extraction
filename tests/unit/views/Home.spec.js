import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home.vue';

describe('Home.vue', () => {
  const shallowMountComponent = () => shallowMount(Home, {
    stubs: ['v-container', 'v-btn', 'v-row', 'v-col'],
  });

  it('renders as expected', () => {
    const wrapper = shallowMountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('extracts a number and render as expected', async () => {
    jest.spyOn(global.Math, 'random')
      .mockReturnValueOnce(0.123456789)
      .mockReturnValueOnce(0.987654321);
    const wrapper = shallowMountComponent();
    wrapper.vm.handleExtract();
    await wrapper.vm.$nextTick();
    expect(wrapper).toMatchSnapshot();
    wrapper.vm.handleExtract();
    await wrapper.vm.$nextTick();
    expect(wrapper).toMatchSnapshot();
    jest.spyOn(global.Math, 'random').mockRestore();
  });
});
