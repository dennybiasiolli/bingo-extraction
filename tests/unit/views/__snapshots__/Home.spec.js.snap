// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders as expected 1`] = `
<div class="home"><img alt="Vue logo" src="../assets/logo.png">
  <helloworld-stub msg="Welcome to Your Vue.js App"></helloworld-stub>
</div>
`;
